### DISCUSSION TITLE
ZLUDA on dev branch offloads to cpu instead of gpu

### DISCUSSION THREAD

**Discussion by dragitz** [vladmandic/sdnext#2869] [Q&A] [answered]
Hello, I know it's not the issue tab but I've read that if there's an issue related to the dev branch, it is better to either discuss it here or on discord. Unfortunately the invite link has expired so I'm making the post here, apologies in advance if it's not the right place :)

First of all, thank you very much for your work and the detailed guide provided here https://github.com/vladmandic/automatic/wiki/ZLUDA

I follwed every step and the last step took around 14 minutes to compile.

After the compilation ended, in the console I saw an incredible boost in it/s
![immagine](https://github.com/vladmandic/automatic/assets/8062792/9b1d2d73-e270-42d1-a6a2-72fc3ac94d6f)

I've checked if ZLUDA was properly installed and it seems to recognize the gpu without any issue
![immagine](https://github.com/vladmandic/automatic/assets/8062792/fd5dd9ab-4e48-4f27-9e35-d01db97783c0)

But after generating the first image the iterations were identical as if I was using ``--use-directml`` combined with my gpu (~2 it/s)
![immagine](https://github.com/vladmandic/automatic/assets/8062792/f1a86e0e-123c-449a-b9bd-19fbc7991791)

I've checked every setting containing the words such as "cpu, gpu or offload" in the system tab but none of them were the issue.
I did notice that in the console it said ``low_cpu_mem_usage: True`` so I took the liberty of editing [this line](https://github.com/vladmandic/automatic/blob/3c952675fefd2c94b817940ffbd4cd94fd5876c9/modules/sd_models.py#L749) to ensure it wasn't the issue, but unfortunately I had no success (this is why it is set to ``False`` in the logs)

Logs:
[sdnext.log](https://github.com/vladmandic/automatic/files/14311969/sdnext.log)

---

**Version Platform Description**

Dev version: 3c952675fefd2c94b817940ffbd4cd94fd5876c9
OS: Windows 10.0.19045 Build 19045
Browser: Firefox 122.0.1
GPU: 6700 xt (gfx1031)
CPU: 5600x

``torch. __version__``  2.2.0+cu118
Extensions enabled: ['Lora', 'sd-extension-chainner', 'sd-extension-system-info', 'sd-webui-agent-scheduler', 'sd-webui-controlnet', 'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg']

SD Model based on 1.5

---

I've also restarted my computer twice just to be sure I've done everything correctly, and my installation is the default one with the exception of the above mentioned edit.
I'm sure something is missing on my side, and I'm struggling to find it, what could the issue be? Does anyone have a similar experience?

**Comment by vladmandic** [comment, marked as answer]
fresh invite: <https://discord.gg/sd-next-federal-batch-inspectors-1101998836328697867>

**Reply by dragitz** [reply]
Thank you very much, I will post it there!

**Reply by LiVeenMusic** [reply]
wtf... what about the rest of us??

**Reply by vladmandic** [reply]
what do you mean? that's a on open invite. plus its in a pinned thread on top of discussions as well.

**Reply by LiVeenMusic** [reply]
You're absolutely correct, pardon me. I was having a bit of a sleep deprived day working on HipBLASLt for my gfx1031 and didn't think straight enough at the time to realize this. Thanks for clarifying, and sorry to bother you!
